modules/oe_whitelabel_search/src/Plugin/Block/SearchBlock.php

```diff
@@ -0,0 +1,261 @@
+      '#title' => $this->t('Button classes'),
+      '#description' => $this->t('Add space-separated classes that will be added to the button.'),
+      '#default_value' => $config['button']['classes'],
+    ];
+    $form['enable_autocomplete'] = [
+      '#type' => 'checkbox',
+      '#title' => $this->t('Enable autocomplete'),
+      '#default_value' => $config['view_options']['enable_autocomplete'] ?? 0,
+    ];
+    $form['view_id'] = [
+      '#type' => 'textfield',
+      '#title' => $this->t('View id'),
+      '#description' => $this->t('The view id will be the machine name for the Search API View.'),
+      '#default_value' => $config['view_options']['id'] ?? '',
+      '#states' => [
+        'visible' => [
+          ':input[name="settings[enable_autocomplete]"]' => ['checked' => TRUE],
+        ],
+      ],
+    ];
+    $form['view_display'] = [
+      '#type' => 'textfield',
+      '#title' => $this->t('View display'),
+      '#description' => $this->t('The view display will be the machine name of the display used, at the views settings in the advanced section.'),
+      '#default_value' => $config['view_options']['display'] ?? '',
+      '#states' => [
+        'visible' => [
+          ':input[name="settings[enable_autocomplete]"]' => ['checked' => TRUE],
+        ],
+      ],
+    ];
+
+    return $form;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function blockSubmit($form, FormStateInterface $form_state): void {
+    $values = $form_state->getValues();
+    $this->setConfigurationValue('form', [
+      'action' => $form_state->getValue('form_action'),
+      'classes' => $form_state->getValue('form_classes'),
+    ]);
+    $input = $values['input'];
+    $this->setConfigurationValue('input', [
+      'name' => $input['input_name'],
+      'label' => $input['input_label'],
+      'classes' => $input['input_classes'],
+      'placeholder' => $input['input_placeholder'],
+    ]);
+    $button = $values['button'];
+    $this->setConfigurationValue('button', [
+      'label' => $button['button_label'],
+      'type' => $button['button_type'],
+      'classes' => $button['button_classes'],
+      'icon' => [
+        'name' => $button['button_icon_name'],
+        'position' => $button['button_icon_position'],
+      ],
+    ]);
+    $this->setConfigurationValue('view_options', [
+      'id' => $form_state->getValue('view_id'),
+      'display' => $form_state->getValue('view_display'),
+      'enable_autocomplete' => $form_state->getValue('enable_autocomplete'),
+    ]);
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function blockValidate($form, FormStateInterface $form_state) {
+    $values = $form_state->getValues();
+    if ($values['enable_autocomplete']) {
+      $view = View::load($values['view_id']);
+      if (!$view || !$view->getDisplay($values['view_display'])) {
+        $form_state->setErrorByName('', t('View id or view display not found.'));
+      }
+    }
+
+    $elements = [];
```

Comment:
dont do this, check each in its own like, dont cycle through them, there are only 2 items and would be much clearere that way